subsys/bluetooth/host/gatt.c

```diff
@@ -1735,6 +1741,26 @@ static u8_t connected_cb(const struct bt_gatt_attr *attr, void *user_data)
 			continue;
 		}
 
+#if defined(CONFIG_BT_SMP)
+		/* Check if attribute requires encryption/authentication */
+		if (attr->perm & BT_GATT_PERM_WRITE_ENCRYPT ||
```

Comment:
```suggestion
		if (attr->perm &( BT_GATT_PERM_WRITE_ENCRYPT | BT_GATT_PERM_WRITE_AUTHEN)) {
```
reason: less assembly instruction in most cases

Reply:
Indeed that would be better, will send an update shortly.

Reply:
I did the update but found out we could reuse the logic for checking the attribute permission for the CCC, so I end up adding another patch.